lib/Sema/TypeCheckDecl.cpp

```diff
@@ -975,6 +976,19 @@ static void checkRedeclaration(TypeChecker &tc, ValueDecl *current) {
           isAcceptableVersionBasedChange = true;
         }
       }
+      // - or they are computed properties of different types,
+      {
+        const auto *currentVD = dyn_cast<VarDecl>(current);
+        const auto *otherVD = dyn_cast<VarDecl>(other);
+        if (currentVD && otherVD &&
+            !currentVD->hasStorage() &&
+            !otherVD->hasStorage() &&
+            !currentVD->getInterfaceType()->isEqual(
+              otherVD->getInterfaceType())) {
```

Comment:
Something like this, maybe?

```swift
class A<T, U> {}

extension A {
  @available(swift 4)
  var x: T? { return nil }
}

extension A where T == U {
  @available(swift, obsoleted: 4)
  var x: U? { return nil }
}
```

I'm just making things up; I don't actually have a test case. It just set off warning bells.